src/models/transferSettlement/facade.js

```diff
@@ -0,0 +1,305 @@
+                  .innerJoin('participantCurrency AS PC1', function () {
+                    this.on('PC1.currencyId', 'PC.currencyId')
+                      .andOn('PC1.participantId', 'PC.participantId')
+                      .andOn('PC1.ledgerAccountTypeId', 'M.settlementAccountTypeId')
+                  })
+                  .where(function () {
+                    this.where({ 'TP.transferId': transferId })
+                    this.andWhere(function () {
+                      this.andWhere({ 'G.name': 'GROSS' })
+                    })
+                  })
+              })
+          }).joinRaw('AS TR ON PP.participantCurrencyId = TR.ParticipantCurrencyId')
+          .transacting(trx)
+
+        // Insert new participant position change records
+        await knex.from(knex.raw('participantPositionChange (participantPositionId, transferStateChangeId, value, reservedValue)'))
+          .insert(function () {
+            this.from('participantPosition AS PP')
+              .select('PP.participantPositionId', 'TSC.transferStateChangeId', 'PP.value', 'PP.reservedValue')
+              .innerJoin(function () {
+                this.from('transferParticipant AS TP')
+                  .select('PC.participantCurrencyId')
+                  .innerJoin('participantCurrency AS PC', 'TP.participantCurrencyId', 'PC.participantCurrencyId')
+                  .innerJoin('settlementModel AS M', 'M.ledgerAccountTypeId', 'PC.ledgerAccountTypeId')
+                  .innerJoin('settlementGranularity AS G', 'M.settlementGranularityId', 'G.settlementGranularityId')
+                  .where(function () {
+                    this.where({ 'TP.transferId': transferId })
+                    this.andWhere(function () {
+                      this.andWhere({ 'G.name': 'GROSS' })
+                    })
+                  })
+                  .union(function () {
+                    this.select('PC1.participantCurrencyId')
+                      .from('transferParticipant AS TP')
+                      .innerJoin('participantCurrency AS PC', 'TP.participantCurrencyId', 'PC.participantCurrencyId')
+                      .innerJoin('settlementModel AS M', 'M.ledgerAccountTypeId', 'PC.ledgerAccountTypeId')
+                      .innerJoin('settlementGranularity AS G', 'M.settlementGranularityId', 'G.settlementGranularityId')
+                      .innerJoin('participantCurrency AS PC1', function () {
+                        this.on('PC1.currencyId', 'PC.currencyId')
+                          .andOn('PC1.participantId', 'PC.participantId')
+                          .andOn('PC1.ledgerAccountTypeId', 'M.settlementAccountTypeId')
+                      })
+                      .where(function () {
+                        this.where({ 'TP.transferId': transferId })
+                        this.andWhere(function () {
+                          this.andWhere({ 'G.name': 'GROSS' })
```

Comment:
'GROSS' should be set using an ENUM from central-services-shared.

Reply:
As discussed with @elnyry-sam-k: flag this issue for a future story If appropriate ENUMS are not already available for this.